packages/portfolio-contract/src/resolver/resolver.exo.ts

```diff
@@ -0,0 +1,253 @@
+            return cctpTransactionRegistry.get(key).vowKit.vow;
+          }
+
+          const vowKit = vowTools.makeVowKit<void>();
+
+          cctpTransactionRegistry.init(
+            key,
+            harden({
+              destinationAddress,
+              amountValue: amount,
+              vowKit,
+            }),
+          );
+
+          trace(`Registered pending CCTP transaction: ${key}`);
+          return vowKit.vow;
+        },
+      },
+      service: {
+        settleCCTPTransaction(
+          chainId: CaipChainId,
+          remoteAddress: `0x${string}`,
+          amountValue: bigint,
+          status: 'confirmed' | 'failed' | 'pending',
+        ) {
+          const destinationAddress: AccountId = `${chainId}:${remoteAddress}`;
+          const key =
+            `${destinationAddress}:${amountValue}` as CCTPTransactionKey;
+
+          if (!cctpTransactionRegistry.has(key)) {
+            trace('No pending CCTP transaction found for key:', key);
+            trace('Available keys in registry:', [
+              ...cctpTransactionRegistry.keys(),
+            ]);
+            throw Error(
+              `No pending CCTP transaction found matching provided details: ${key}`,
+            );
+          }
+
+          const registryEntry = cctpTransactionRegistry.get(key);
+
+          switch (status) {
+            case 'confirmed':
+              trace(`CCTP transaction confirmed - resolving pending operation`);
+              registryEntry.vowKit.resolver.resolve();
+              cctpTransactionRegistry.delete(key);
+              return;
+
+            case 'failed':
+              trace(`CCTP transaction failed - rejecting pending operation`);
+              registryEntry.vowKit.resolver.reject(
+                Error('CCTP transaction failed'),
+              );
+              cctpTransactionRegistry.delete(key);
+              return;
+
+            case 'pending':
+              trace(`CCTP transaction still pending - no action taken`);
+              return;
+          }
+        },
+      },
+      debug: {
+        getPendingTransactions() {
+          return [...cctpTransactionRegistry.keys()];
+        },
+      },
+    },
+  );
+};
+
+harden(prepareCCTPResolver);
+
+/**
+ * Prepare CCTP Resolver Invitation Makers.
+ *
+ * @param zone - Durable storage zone
+ * @param zcf - Zoe Contract Facet for creating invitations
+ * @param resolver - The resolver instance to use
+ * @returns A function to create resolver invitation makers
+ */
+export const prepareResolverInvitationMakers = (
+  zone: Zone,
+  zcf: ZCF,
+  resolver: CCTPResolverServiceFacet,
+) => {
+  const resolverInvitationMakersZone = zone.subZone('ResolverInvitationMakers');
+
+  // Prepare durable settlement handler
+  const prepareSettlementHandler = resolverInvitationMakersZone.exoClass(
+    'SettlementHandler',
+    M.interface('SettlementHandler', {
+      handle: M.call(M.any(), M.any()).returns(M.promise()),
```

Comment:
To guard the return of an async method, use `callWhen`:

```suggestion
      handle: M.callWhen(M.any(), M.any()).returns(...a specific shape here...),
```